### PR TITLE
osdc: fix honor_osdmap_full

### PR DESCRIPTION
**Pull request by jcsp**
This was being ignored in new code that tested
the per-pool full flags.  Refactor it so that
all the tests of full flags go through _osdmap_pool_full
and thereby experience the honor_osdmap_full policy.

Fixes: #12971
Signed-off-by: John Spray <john.spray@redhat.com>

### CODE REVIEW

File: src/osdc/Objecter.cc [hunk 1]
```diff
@@ -2415,19 +2417,24 @@ bool Objecter::_osdmap_pool_full(const int64_t pool_id) const
     return false;
   }
 
-  return pool->has_flag(pg_pool_t::FLAG_FULL);
+  return _osdmap_fpool_full(*pool);
```
Comment (gregsfortytwo) [inline src/osdc/Objecter.cc]: typo ;)

**Comment by gregsfortytwo** [comment]
Rest looks good, grep says you got them all.

**Comment by jcsp** [comment]
Added a commit for the OSD, which also had a new pool flag check that wasn't taking account of the old mds special casing.

**Comment by gregsfortytwo** [comment]
Looks good, waiting for build to succeed.

**Pull request merged**